src/error.rs

```diff
@@ -0,0 +1,32 @@
+use std::error;
+use std::fmt;
+
+/// Error type for the library.
+#[derive(Clone,Debug)]
+pub struct Error {
```

Comment:
I was thinking of using `error_chain` at some stage in the future, but for now this is definitely a move in the right direction.